lms/static/scripts/frontend_apps/components/SubmitGradeForm.js

```diff
@@ -35,39 +41,54 @@ const useFetchGrade = (student, setFetchGradeError) => {
   const gradingService = useService(GradingService);
   const [grade, setGrade] = useState('');
   const [gradeLoading, setGradeLoading] = useState(false);
-
-  useEffect(() => {
-    /** @type {boolean} */
-    let ignoreResults;
-    if (student) {
-      // Fetch the grade from the service api
-      // See https://www.robinwieruch.de/react-hooks-fetch-data for async in useEffect
-      const fetchData = async () => {
-        setGradeLoading(true);
-        setGrade(''); // Clear previous grade so we don't show the wrong grade with the new student
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debouncedFetchGrade = useCallback(
+    debounce(
+      /**
+       * @param {StudentInfo} student
+       * @param {[boolean]} keepResult
+       */
+      async (/** @type {StudentInfo} */ student, keepResult) => {
         try {
           const { currentScore } = await gradingService.fetchGrade({ student });
-          if (!ignoreResults && currentScore) {
-            // Only set these values if we didn't cancel this request
+          // Only set these values if we didn't cancel this request
+          if (keepResult[0]) {
+            if (currentScore === null) {
+              setGrade('');
+              return;
+            }
             setGrade(scaleGrade(currentScore, GRADE_MULTIPLIER));
           }
         } catch (e) {
           setFetchGradeError(e);
         } finally {
           setGradeLoading(false);
         }
-      };
-      fetchData();
+      },
+      200
+    ),
+    []
+  );
+
+  useEffect(() => {
+    /** @type [boolean] */
+    const keepResult = [true];
```

Comment:
keepResult needs to be an Array or an Object, because the value of this variable could change during the lifetime of the debounced function execution. We pass by reference to the debounce function, so the reference leads to the current value, when it is needed.